From 6b4d889d692e830f6001ce417392d9353cc56dc4 Mon Sep 17 00:00:00 2001
From: Sean Risser <srisser@google.com>
Date: Thu, 20 Jan 2022 03:35:37 -0500
Subject: [PATCH] Delay initializing fragment_shading_rate tests

When the shading rate tests were renamed, they turned a delayed call
to createChildren into an immediate call. For partial test-suite runs,
this can significantly impact run times. So the

Components: Vulkan

VK-GL_CTS Issue: 3460

Affects:
dEQP-VK.fragment_shading_rate.*

Change-Id: Ie2bb7a7690b2650006da888433397dae9a5ff7cf
---

diff --git a/external/vulkancts/modules/vulkan/fragment_shading_rate/vktFragmentShadingRateTests.cpp b/external/vulkancts/modules/vulkan/fragment_shading_rate/vktFragmentShadingRateTests.cpp
index cfd53eb..6e1e0ec 100644
--- a/external/vulkancts/modules/vulkan/fragment_shading_rate/vktFragmentShadingRateTests.cpp
+++ b/external/vulkancts/modules/vulkan/fragment_shading_rate/vktFragmentShadingRateTests.cpp
@@ -424,8 +424,9 @@
 	parentGroup->addChild(group.release());
 }
 
-void createChildren (tcu::TestContext& testCtx, tcu::TestCaseGroup* group, bool useDynamicRendering)
+void createChildren (tcu::TestCaseGroup* group, bool useDynamicRendering)
 {
+	tcu::TestContext&	testCtx		= group->getTestContext();
 	createBasicTests(testCtx, group, useDynamicRendering);
 	createAttachmentRateTests(testCtx, group, useDynamicRendering);
 
@@ -444,11 +445,8 @@
 tcu::TestCaseGroup* createTests (tcu::TestContext& testCtx)
 {
 	de::MovePtr<tcu::TestCaseGroup> mainGroup				(new tcu::TestCaseGroup(testCtx, "fragment_shading_rate", "Fragment shading rate tests"));
-	de::MovePtr<tcu::TestCaseGroup> renderpass2Group		(new tcu::TestCaseGroup(testCtx, "renderpass2", "Draw using render pass object"));
-	de::MovePtr<tcu::TestCaseGroup> dynamicRenderingGroup	(new tcu::TestCaseGroup(testCtx, "dynamic_rendering", "Draw using VK_KHR_dynamic_rendering"));
-
-	createChildren(testCtx, renderpass2Group.get(), false);
-	createChildren(testCtx, dynamicRenderingGroup.get(), true);
+	de::MovePtr<tcu::TestCaseGroup> renderpass2Group		(createTestGroup(testCtx, "renderpass2", "Draw using render pass object", createChildren, false));
+	de::MovePtr<tcu::TestCaseGroup> dynamicRenderingGroup	(createTestGroup(testCtx, "dynamic_rendering", "Draw using VK_KHR_dynamic_rendering", createChildren, true));
 
 	mainGroup->addChild(renderpass2Group.release());
 	mainGroup->addChild(dynamicRenderingGroup.release());
